### PR TITLE
(BKR-280) ability to install pe/foss on subset of hosts

### PR DESCRIPTION
**Pull request by anodelman**
- created install_pe_on, upgrade_pe_on, install_puppet_on that accept a
  first argument of a host array and then correctly install pe/puppet on
  provided hosts

### CODE REVIEW

**Comment by anodelman** [comment]
retest this please

**Comment by anodelman** [comment]
retest this please

**Comment by anodelman** [comment]
Fix for broken beaker-rspec testing here:

https://github.com/puppetlabs/beaker-rspec/pull/69

**Comment by anodelman** [comment]
retest this please

File: spec/beaker/dsl/structure_spec.rb [hunk 1]
```diff
@@ -128,6 +128,14 @@ class ClassMixedWithDSLStructure
       subject.confine( :to, {} )
     end
 
+    it 'uses a provided host subset when no criteria is provided' do
+
+      subset = ['host1', 'host2']
+      hosts = subset << 'host3'
```
Comment (kevpl) [inline spec/beaker/dsl/structure_spec.rb]: @anodelman it looks like `hosts` and `subset` are the same after this point, making this test not as helpful:

    ➜  beaker git:(master) ✗ ruby --version
    ruby 1.9.3p484 (2013-11-22 revision 43786) [x86_64-darwin13.0.0]
    ➜  beaker git:(master) ✗ irb
    irb(main):001:0> s1 = [1, 2]
    => [1, 2]
    irb(main):002:0> s2 = s1 << 3
    => [1, 2, 3]
    irb(main):003:0> s2
    => [1, 2, 3]
    irb(main):004:0> s1
    => [1, 2, 3]
    irb(main):005:0> s2.push(4)
    => [1, 2, 3, 4]
    irb(main):006:0> s1
    => [1, 2, 3, 4]

`Array.new` or `compact` look to be good solutions to this issue from my experiments.  Maybe I'm misunderstanding what you're going for here?

**Comment by anodelman** [comment]
Failures unrelated to patch, repaired spec test.

**Pull request merged**